contrib/inventory/azure_rm.py

```diff
@@ -361,12 +366,24 @@ def _get_env_credentials(self):
 
         return None
 
+    def _get_azure_cli_credentials(self):
+        credentials, subscription_id = get_azure_cli_credentials()
```

Comment:
For this to work, you need to add "azure-cli-core" as dependencies. This is not a dependencies of "azure-common" because:
1- this would create a circular PyPI dependencies
2- it's documented as a helper for people that have the CLI installed.

Reply:
I copy this logic from [`aure_rm_common.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L573). I don't understand what is the interface from azure-cli-core? I notice that the [`get_azure_cli_credentials()`](https://github.com/Azure/azure-sdk-for-python/blob/master/azure-common/azure/common/credentials.py#L31) do several things and retrieve the subscriptions and credential.

Reply:
Have a closer look at `azure_rm_common.py`, while we import from `azure-cli-core`, it is done conditionally and the script does not require it to work. If the package is present then cli auth will work but if it is not then either cli auth is skipped or it will fail saying you need to install `azure-cli-core`. We do this by putting the following in the import area

```
try:
    from azure.cli.core.util import CLIError
    from azure.common.credentials import get_azure_cli_credentials, get_cli_profile
    from azure.common.cloud import get_cli_active_cloud
except ImportError:
    HAS_AZURE_CLI_CORE = False
    CLIError = Exception
```

When attempting to get the cli credentials we put a check on `HAS_AZURE_CLI_CORE` and work with that. See https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L529-L537 for more info.

On a side note, you may be better off just removing the changes in the inventory script and put it in another PR. I don't see a reason why we need to bundle the MSI changes with this inventory change.